demos/common/pipelines/include/pipelines/async_pipeline.h

```diff
@@ -57,14 +58,16 @@ class AsyncPipeline {
     virtual int64_t submitData(const InputData& inputData, const std::shared_ptr<MetaData>& metaData);
 
     /// Gets available data from the queue
-    /// Function will treat results as ready only if next sequential result (frame) is ready.
-    virtual std::unique_ptr<ResultBase> getResult();
+    /// @param shouldKeepOrder if true, function will treat results as ready only if next sequential result (frame) is
+    /// ready (so results can be extracted in the same order as they were submitted). Otherwise, function will return if any result is ready.
+    virtual std::unique_ptr<ResultBase> getResult(bool shouldKeepOrder = true);
 
 protected:
     /// Returns processed result, if available
-    /// Function will treat results as ready only if next sequential result (frame) is ready.
+    /// @param shouldKeepOrder if true, function will return processed data sequentially,
+    /// keeping original frames order (as they were submitted). Otherwise, function will return processed data in random order.
     /// @returns InferenceResult with processed information or empty InferenceResult (with negative frameID) if there's no any results yet.
-    virtual InferenceResult getInferenceResult();
+    virtual InferenceResult getInferenceResult(bool shouldKeepOrder);
 
     std::unique_ptr<RequestsPool> requestsPool;
     std::unordered_map<int64_t, InferenceResult> completedInferenceResults;
```

Comment:
we might want to change this to map<> to avoid delays of some items in case of shouldKeepOrder=false.
It's quite low-probable situation, but in case of fast filling completedInferenceResults with items and shouldKeepOrder=false some items might be processed with large delay as item will be popped out of queue according to hash value and extraction order will not correspond to pushing order. In case of map<> values will be ordered according to frame number - that will keep FIFO order of already processed frames.
However, in our demos we're emptying whole queue of completed results before submitting new ones - in this everything should be fine even with unordered_map (and it is faster than map).
Still need to discuss that.